src/common/translations/localize.ts

```diff
@@ -116,13 +94,21 @@ export const computeLocalize = async <Keys extends string = LocalizeKeys>(
       } catch (err: any) {
         return "Translation error: " + err.message;
       }
-      cache._localizationCache[messageKey] = translatedMessage;
+      localizationCache[translatedValue] = translatedMessage;
     }
 
     let argObject = {};
+    // TS says that we will always get the new format, but we might get the old format
+    const args = _args as any;
     if (args.length === 1 && typeof args[0] === "object") {
       argObject = args[0];
-    } else {
+      if (Object.values(argObject).some((v) => v === undefined)) {
+        // eslint-disable-next-line no-console
+        console.warn("[FIXME] While localizing", key, "undefined was passed");
+      }
+    } else if (args.length >= 2) {
+      // eslint-disable-next-line no-console
+      console.warn("[FIXME] While localizing", key, "old format was passed");
```

Comment:
We should link devs to a dev blog post what to do about this.

Reply:
Good idea. Should I create a PR at https://github.com/home-assistant/developers.home-assistant?